crates/std/src/uncategorized/asset_url.rs

```diff
@@ -436,11 +436,20 @@ impl AssetUrl {
             AssetUrl::Absolute(url) => Ok(Self::Absolute(url.join(path)?)),
             AssetUrl::Relative(p) => Ok(Self::Relative(
                 // The Url::join method has some intricacies so we want these to behave the same
-                Url::parse(&format!("http://localhost/{}", p.as_str()))
+                if cfg!(target_os = "windows") {
```

Comment:
sadly on windows, when using a pipeline.json file that contains more than one pipeline like this : 

`
[
    {
        "pipeline": {
            "type": "Materials",
            "importer": {
            "type": "Single",
            "name": "sand",
            "base_color": "sand.png"
            },
            "output_quads": true
        }
    },
    {
        "pipeline": {
          "type": "Materials",
          "importer": {
            "type": "Single",
            "name": "bark",
            "base_color": "bark.png"
          },
          "output_quads": true
        }
      }
]
`

app fails at runtime because it can't save the material asset(s) file ie : c:\\ambient\\guest\\rust\\examples\\advanced\\forest\\build\\assets\\pipeline.json/0/mat.json

because path mixes slashes and backslashes

i've sumbitted a PR on udoprog/relative-path here https://github.com/udoprog/relative-path/pull/43/files

so we don't maintain this platform-specific code